modules/swagger-codegen/src/main/resources/lua/configuration.mustache

```diff
@@ -0,0 +1,43 @@
+{{>partial_header}}
+package {{packageName}}
+
+import (
+	"net/http"
+)
+
+const ContextOAuth2 		int = 1
+const ContextBasicAuth 		int = 2
+const ContextAccessToken 	int = 3
+const ContextAPIKey 		int = 4 
+
+type BasicAuth struct {
+	UserName      string            `json:"userName,omitempty"`
+	Password      string            `json:"password,omitempty"`	
+}
+
+type APIKey struct {
+	Key 	string
+	Prefix	string
+}
+
+type Configuration struct {
```

Comment:
@daurnimator here is the configuration class for Go API client. We may need something similar in Lua API client to store the authentication setting, default header, host, basePath, etc